interview-questions.md

```diff
@@ -8,62 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: The map method iterates over an array calling a function each time it iterates over an index of the array. This method is used to perform some action on the original array and returns data to a new array. map does not change the original array. The filter method also calls a function while itereating ove each index of an array, however it is testing for certain criteria of the data. If the data equals true then it is stored in a new array which can be accessed and returned.
 
-  Researched answer:
+  Researched answer: map is a method that creates a new array by changing each index element individually, whill the filter method creates an array by removing elements that do not fit the current conditions.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is a set block of code that can be reused creating the ability to not have to rewrite a program everyime you want it to run. A method on the other hand has properties that can can be accessed to perform a specific action and can be accesed using dot notation.
 
-  Researched answer:
+  Researched answer: A function is a reusable block of code that is invoked by its name. It can pass and return data that it operates on. A method is called by its name that is associated with an object. A method is an instance of a class.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is the prcocess of taking values from objects and arrays and storing them in a variable. This allows faster and easier access to this information to us in a program. Instead of using dot or bracket notation to continuously access nested arrays, objects or just values you can call the variable name to gain all of the information you have stored. The path to the information such as <class.array[0]> will all be stored in the variable such as<path>.
 
-  Researched answer:
+  Researched answer: Object destructing is a Javascript expresion that allows you to extract data from arrays and objects. It hen stores it in a variable which can be called by name to by accessed later.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Hoisting is the act of moving the  declaration of functions, variables and classes higher in their scope befor the code is executed. This can be safely done with functions so that you may use them prior to them being executed in the code. With variables and classes the hoisted declarations are not ran until the code is executed.
 
-  Researched answer:
+  Researched answer: A proceess where the interpreter appears to move declarations of functions, classes and variables to the top of its scope befor the execution of its code.
```

Comment:
Incredible answers in this section!  Thank you for the hard work you put into this.